src/main/java/org/jabref/gui/cleanup/CleanupSingleAction.java

```diff
@@ -0,0 +1,105 @@
+package org.jabref.gui.cleanup;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.swing.undo.UndoManager;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.actions.ActionHelper;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.undo.NamedCompound;
+import org.jabref.gui.undo.UndoableFieldChange;
+import org.jabref.logic.cleanup.CleanupWorker;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.FieldChange;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.preferences.CleanupPreferences;
+import org.jabref.preferences.PreferencesService;
+
+public class CleanupSingleAction extends SimpleCommand {
+
+    private final PreferencesService preferences;
+    private final DialogService dialogService;
+    private final StateManager stateManager;
+    private final BibEntry entry;
+    private final UndoManager undoManager;
+
+    private boolean isCanceled;
+    private int modifiedEntriesCount;
+
+    public CleanupSingleAction(BibEntry entry, PreferencesService preferences, DialogService dialogService, StateManager stateManager, UndoManager undoManager) {
+        this.entry = entry;
+        this.preferences = preferences;
+        this.dialogService = dialogService;
+        this.stateManager = stateManager;
+        this.undoManager = undoManager;
+
+        this.executable.bind(ActionHelper.needsEntriesSelected(stateManager));
+    }
+
+    @Override
+    public void execute() {
+        isCanceled = false;
```

Comment:
Although the executable binding is set, the executability of the command is not dependend on it. The executable property is just for the ui control to enable or disable it. An unknowing developer could still create the command in the code and execute it. So I believe an additional check for the requirements to be met is needed at the beginning of the executable command.

Reply:
In the file `CleanupAction.java`, there are additional checks at the beginning of the executable command:
```
if (stateManager.getActiveDatabase().isEmpty()) {
            return;
        }

if (stateManager.getSelectedEntries().isEmpty()) { // None selected. Inform the user to select entries first.
            dialogService.showInformationDialogAndWait(Localization.lang("Cleanup entry"), Localization.lang("First select entries to clean up."));
            return;
        }
```
However, it seems that these requirements are always met for an entry based cleanup once the entryeditor is opened.
What kinds of additional checks should I add?
Could you please give more details?

Reply:
Ooops, missed that. Ok, then it looks good for me